src/main/java/com/linecorp/armeria/client/routing/NodeSelector.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.routing;
+
+
+import java.util.List;
+
+public interface NodeSelector<T extends Node> {
+    /**
+     * Return the {@link ServerGroup} this selector hold
+     */
+    ServerGroup<T> serverGroup();
+
+    /**
+     * Return a selected node.
+     */
+    Node select(List<T> nodes);
```

Comment:
- `select()` method does not need a parameter given that `NodeSelector` already has a reference to `ServerGroup`.
- Could we return `T` instead of `Node`?